Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/LibraryBuilder.java

```diff
@@ -326,31 +334,51 @@ else if (e.getSeverity() == CqlTranslatorException.ErrorSeverity.Info) {
         }
     }
 
+    private boolean shouldReport(CqlTranslatorException.ErrorSeverity errorSeverity) {
+        switch (errorLevel) {
+            case Info:
+                return
+                        errorSeverity == CqlTranslatorException.ErrorSeverity.Info
+                                || errorSeverity == CqlTranslatorException.ErrorSeverity.Warning
+                                || errorSeverity == CqlTranslatorException.ErrorSeverity.Error;
+            case Warning:
+                return
+                        errorSeverity == CqlTranslatorException.ErrorSeverity.Warning
+                                || errorSeverity == CqlTranslatorException.ErrorSeverity.Error;
+            case Error:
+                return errorSeverity == CqlTranslatorException.ErrorSeverity.Error;
+            default:
+                throw new IllegalArgumentException(String.format("Unknown error severity %s", errorSeverity.toString()));
+        }
+    }
```

Comment:
I looked at using the ordinal() when I implemented this, because you're right, it could be replaced by that code, but since I didn't assign the ordinal values to the enums, that felt very much like relying on an implementation detail to describe the behavior. It's also my understanding that best practice is not to use the ordinal values, so I opted against using them. Do you agree?